src/main/java/org/mastodon/mamut/launcher/MastodonLauncher.java

```diff
@@ -654,4 +634,49 @@ private WindowManager createWindowManager()
 	{
 		return new WindowManager( context );
 	}
+
+	private void openMastodonMainWindow( @Nonnull File file )
+	{
+		try
+		{
+			final WindowManager windowManager = createWindowManager();
+			final MamutProject project = new MamutProjectIO().load( file.getAbsolutePath() );
+			windowManager.getProjectManager().open( project, true );
+			new MainWindow( windowManager ).setVisible( true );
+			dispose();
+			/*
+			 * We update the list of recent projects here so that
+			 * only projects that were successfully opened are added
+			 * to the list.
+			 */
+			RecentProjectsPanel.recentProjects.add( file.getAbsolutePath() );
+		}
+		catch ( final IOException | SpimDataException e )
+		{
+			gui.error( "Invalid Mastodon file.\n\n" + LauncherUtil.toMessage( e ) );
+		}
+	}
+
+	private class MastodonDropTarget extends DropTarget
+	{
+		@Override
+		public synchronized void drop( DropTargetDropEvent dropTargetDropEvent )
+		{
+			try
+			{
+				dropTargetDropEvent.acceptDrop( DnDConstants.ACTION_COPY );
+				@SuppressWarnings( "unchecked" )
+				List< File > droppedFiles = ( List< File > ) dropTargetDropEvent.getTransferable().getTransferData( DataFlavor.javaFileListFlavor );
+				for ( File file : droppedFiles )
+				{
+					// process files
+					openMastodonMainWindow( file );
```

Comment:
loadMastodonProject( file.getAbsolutePath() ) would also work here. What's the reasoning of having the openMastodonMainWindow method.

Reply:
I agree.
I kept the original loadMastodonProject-method, since in another test, it worked also with this method.